include/seahorn/Transforms/Instrumentation/CrabAnalysis.hh

```diff
@@ -0,0 +1,43 @@
+#ifndef _CRAB_ANALYSIS_HH_
+#define _CRAB_ANALYSIS_HH_
+
+#include "llvm/IR/Module.h"
+
+namespace llvm {
+class TargetLibraryInfoWrapperPass;
+} // namespace llvm
+
+namespace seadsa {
+class GlobalAnalysis;
+} // namespace seadsa
+
+namespace clam {
+class CrabBuilderManager;
+class InterGlobalClam;
+} // namespace clam
+
+namespace seahorn {
+using namespace llvm;
+
+class CrabAnalysis {
+  //// \brief crab's cfg builder manager
+  std::unique_ptr<clam::CrabBuilderManager> m_cfg_builder_man;
+  //// \brief crab instance to solve alloc bounds
+  std::shared_ptr<clam::InterGlobalClam> m_crab;
```

Comment:
why a `shared_ptr`? You can use `unique_ptr` and return `clam::InterGlobalClam&` in `getCrab`